### PR TITLE
Updated and refined documentation

### LINKED ISSUE #2753: Docs: Update the Platform specification's "Hardware Folders structure"
### Describe the request

https://arduino.github.io/arduino-cli/1.1/platform-specification/ describes the platform specification. https://arduino.github.io/arduino-cli/1.1/platform-specification/#hardware-folders-structure describes its folder structure.

This folder structure does not correlate with what is seen when installing a specific platform via the board manager on Windows 11 in the Arduino IDE.

I wish for the documentation to be updated to be correct for how it looks in a installation on Windows

### Describe the current behavior

I asked Perplexity why I saw this difference between my installation and what the platform specification said, and got a pretty sold answer I believe.

## My question

```text
The Arduino platform specification says the directories look like:

hardware/arduino/avr/...     - Arduino - AVR Boards
hardware/arduino/sam/...     - Arduino - SAM (32bit ARM) Boards
hardware/yyyyy/avr/...       - Yyy - AVR
hardware/xxxxx/avr/...       - Xxx - AVR

But when I check in explorer in Windows 11, it looks like:

packages\arduino\hardware\avr\1.8.6\cores\arduino

I don't understand why.
```

## The answer
Image

### Arduino CLI version

1.1.0

### Operating system

Windows

### Operating system version

11

### Additional context

The entire nomenclature around Arduino "platform" and "core" is confusing. There are numerous Stack Overflow, forum and what not questions out there about this. For example even this repo has this issue https://github.com/arduino/arduino-cli/issues/549.

Seeing this folder structure difference completely adds to this confusion.

### Issue checklist

- [x] I searched for previous requests in [the issue tracker](https://github.com/arduino/arduino-cli/issues?q=)
- [x] I verified the feature was still missing when using the [nightly build](https://arduino.github.io/arduino-cli/dev/installation/#nightly-builds)
- [x] My request contains all necessary details

### PR DESCRIPTION
**Pull request by cmaglie**
## Please check if the PR fulfills these requirements

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

- [X] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [X] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)
- [ ] `configuration.schema.json` updated if new parameters are added.

## What kind of change does this PR introduce?

* Refined platform specification on installation folder for platforms
* Expanded docs on default directories configuration
* Cleared usage of identification properties

## Other information

Fix #2753

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -8,7 +8,8 @@
   - `data` - directory used to store Boards/Library Manager index files and Boards Manager platform installations.
   - `downloads` - directory used to stage downloaded archives during Boards/Library Manager installations.
   - `user` - the equivalent of the Arduino IDE's ["sketchbook" directory][sketchbook directory]. Library Manager
-    installations are made to the `libraries` subdirectory of the user directory.
+    installations are made to the `libraries` subdirectory of the user directory. Users can manually install 3rd party
+    platforms in the `hardware` subdirecotry of the user directory.
```
Comment (alessio-perugini) [inline docs/configuration.md:12]:
```suggestion
    platforms in the `hardware` subdirectory of the user directory.
```
typo

File: docs/platform-specification.md [hunk 2]
```diff
@@ -5,31 +5,62 @@ Platforms add support for new boards to the Arduino development software. They a
...
+{directories.user}/hardware/bar/avr/...
 ```
 
-The vendor "arduino" has two supported architectures (AVR and SAM), while "xxxxx" and "yyyyy" have only AVR.
+In this latter case the version is omitted.
+
+We can also see that the vendor "arduino" has two supported architectures (AVR and SAM), while "foo" and "bar" have only
+AVR.
+
+### Notes about the choosing the architecture name
```
Comment (alessio-perugini) [inline docs/platform-specification.md:55]:
```suggestion
### Notes about choosing the architecture name
```

**Review by alessio-perugini — CHANGES_REQUESTED** [review]

**Review by alessio-perugini — APPROVED** [review]

File: docs/platform-specification.md [hunk 3]
```diff
@@ -5,31 +5,62 @@ Platforms add support for new boards to the Arduino development software. They a
...
+
+A vendor/maintainer can have multiple supported architectures.
+
+For example, below we have four platforms downloaded from three hypothetical hardware vendors called "arduino", "foo"
+and "bar", and installed using the Board Manager:
+
+```
+{directories.data}/packages/arduino/hardware/avr/1.2.3/...
+{directories.data}/packages/arduino/hardware/sam/1.4.5/...
+{directories.data}/packages/foo/hardware/avr/1.0.2/...
```
Comment (thernstig) [inline docs/platform-specification.md:37]: Would it make more sense to use something else than "avr" here (and below)? It looks strange when you have three hardware vendors, but all three manufacture the same "architecture" avr, as avr is specific only to a specific vendor.

Use some other vendor/architecture combination instead for the foo and bar examples.

**Comment by thernstig** [comment]
I also noticed that if I install a platform manually in `{directories.user}/hardware/STMicroelectronics/stm32/...`, while already having a installation via the Board Manager in `{directories.data}/packages/STMicroelectronics/hardware/stm32/2.8.1/...`, the one in the user directory takes precedence for what is seen in the Arduino IDE.

Example:

image

The problem is however, that I had a new version in ``{directories.user}/hardware/STMicroelectronics/stm32/...`` with new board variants, but I could not see those in the `Board part number: ...` menu as expected.

Is this a separate bug? In that it is possible to install a platform manually, and it is seen in the menu, but the actual new additions to the manual installation are not shown?

Also it might be good to mention in these docs what happens if you have the same `VENDOR_NAME` and `ARCHITECTURE` in these data and user directory?

File: docs/platform-specification.md (hunk 3 above)
Comment (cmaglie) [reply docs/platform-specification.md:37]: I've updated the PR with a real example (with more architecture/vendors combination)

**Comment by cmaglie** [comment]
> Is this a separate bug? In that it is possible to install a platform manually, and it is seen in the menu, but the actual new additions to the manual installation are not shown?

Currently, if there are two matching platforms the one installed in the sketchbook should take precedence... but since you're not able to see the new board definitions there could be a regression, I'll check this out.
To avoid confusion, during the development of a platform, we use the "trick" to slightly rename the vendor installed in the sketchbook so instead of:

`{directories.user}/hardware/STMicroelectronics/stm32/...`

we use for example:

`{directories.user}/hardware/STMicroelectronics-dev/stm32/...`

This way there is no confusion and both platforms (the original and the "dev") are visible in the IDE.

> Also it might be good to mention in these docs what happens if you have the same VENDOR_NAME and ARCHITECTURE in these data and user directory?

Yes good point.

**Comment by thernstig** [comment]
> This way there is no confusion and both platforms (the original and the "dev") are visible in the IDE.

This was not the case for me. I only saw the one as you see in the picture above.

File: docs/platform-specification.md [hunk 4]
```diff
@@ -26,31 +26,34 @@ Alternatively, a platform may be manually installed by the user inside the Sketc
...
+{directories.data}/packages/arduino/hardware/avr/1.8.6/...
+{directories.data}/packages/arduino/hardware/esp32/2.0.18-arduino.5/...
+{directories.data}/packages/arduino/hardware/nrf52/1.4.5/...
+{directories.data}/packages/adafruit/hardware/nrf52/1.6.1/...
+{directories.data}/packages/esp32/hardware/esp32/3.0.7/...
 ```
 
-If the same platform were manually installed by the user, he should have unpacked them inside the following directories:
+In this example three architectures have been installed from the vendor `arduino` (`avr`, `esp32` and `nrf52`), and one
+from `adafruit` and `esp32` (`nrf52` and `esp32` respectively). Note that the vendor `esp32` has the same name of the
```
Comment (thernstig) [inline docs/platform-specification.md:43]: nit: has the same name of the -> has the same name as the

**Review by thernstig — COMMENTED** [review]
LGTM (except I saw no mention of what happens there are matching VENDOR+ARCHITECTURE namings in the data and user directory.

**Comment by cmaglie** [comment]
> except I saw no mention of what happens there are matching VENDOR+ARCHITECTURE namings in the data and user directory

I checked, and the Arduino CLI has a bug/regression. The expected behavior is not happening (the platform installed in the sketchbook should take priority over the one installed via Board Manager if they match VENDOR+ARCHITECTURE).

I'm going to open a new issue about this, I'll add the note in the documentation after the bug is fixed.
In the meantime, I'm merging this one.

**Pull request merged**